src/toggle/toggle.tsx

```diff
@@ -122,6 +185,177 @@ export const Toggle = ({
+                        paddingTop:
+                            !selected || errorMssgOrChildrenShown
+                                ? "0rem"
+                                : "0.6875rem",
+                    }}
+                ></div>
+                <ViewMoreOrLessButtonContainer
+                    $show={!collapsible ? false : selected}
+                    $disabled={disabled}
+                    onClick={() => handleView()}
+                    data-testid="toggle-button"
+                >
+                    <ViewMoreOrLessButtonLabel
+                        weight="semibold"
+                        $disabled={disabled}
+                        data-testid="toggle-button-label"
+                    >
+                        {showMore ? "Show less" : "Show more"}
+                    </ViewMoreOrLessButtonLabel>
+                    {showMore ? (
+                        <ChevronUpIcon height={18} width={18} />
+                    ) : (
+                        <ChevronDownIcon height={18} width={18} />
+                    )}
+                </ViewMoreOrLessButtonContainer>
+            </>
+        );
+    };
+
+    const renderToggleWithRemoveButton = () => {
+        return (
+            <HeaderContainer
+                id={`header-container-${id}`}
+                $disabled={disabled}
+                $error={error}
+                $selected={selected}
+                $indicator={indicator}
+                $styleType={styleType}
+            >
+                <IndicatorLabelContainer $addPadding={removable}>
+                    <Input
+                        ref={inputRef}
+                        name={name}
+                        id={generatedInputId}
+                        type={type === "checkbox" ? "checkbox" : "radio"}
+                        data-testid="toggle-input"
+                        disabled={disabled}
+                        onChange={handleOnChange}
+                        checked={selected}
+                    />
+                    {indicator && renderIndicator()}
+                    <TextContainer>
+                        <Label
+                            htmlFor={generatedInputId}
+                            $selected={selected}
+                            $indicator={indicator}
+                            $disabled={disabled}
+                            data-testid={`toggle-label-${id}`}
+                        >
+                            {children}
+                        </Label>
+                        {subLabel && renderSubLabel()}
+                    </TextContainer>
+                </IndicatorLabelContainer>
+
+                {removable && (
+                    <>
+                        <ButtonContainer
+                            $disabled={disabled}
+                            onClick={handleOnRemove}
+                            id={`remove-${id}`}
+                        >
+                            <ButtonLabel $disabled={disabled}>
```

Comment:
what's the use of the label element nested in a button? can this be directly rendered as `<RemoveButton>Remove</RemoveButton>`?